source/core/solid_base.cc

```diff
@@ -104,12 +190,16 @@ SolidBase<dim, spacedim>::setup_triangulation(const bool restart)
   // Refine the solid triangulation to its initial size
   // NB: solid_tria should not be refined if loaded from a restart file
   // afterwards
-  if (!restart)
+  if (!restart && !param->solid_mesh.simplex)
     {
       solid_tria->refine_global(param->solid_mesh.initial_refinement);
       // Initialize dof handler for solid
-      FE_Q<dim, spacedim> fe(1);
-      solid_dh.distribute_dofs(fe);
+      solid_dh.distribute_dofs(*fe);
+    }
+  else if (!restart)
```

Comment:
Same as before. This says that `!restart` is the logical complement of `!restart && !param->solid_mesh.simplex`, when really here you have two sub-cases for a `!restart` condition. This decreases the code readability in my point of view. Of course now the conditions are close enough to each other so that we can get it, but it might not be the case in the future. So I would prefer :
```
if (!restart)
  if(!param->solid_mesh.simplex)
    {...}
  else
    {...}
```